src/Compiler/Service/ServiceParsedInputOps.fs

```diff
@@ -972,11 +984,60 @@ module ParsedInput =
                                     Some (CompletionContext.ParameterList args)
                                 | _ -> 
                                     defaultTraverse expr
+
+                            | SynExpr.Match (clauses = clauses)
+                            | SynExpr.MatchBang (clauses = clauses) ->
+                                let rec traverse pos clausePat =
```

Comment:
Please factor this whole clause out to a separate top-level function at the same level as `TryGetCompletionContext `.  I'd eventually like to see all the inner functions of `TryGetCompletionContext` flattened out with much less nesting.